crates/operator/src/lib.rs

```diff
@@ -52,15 +53,29 @@ impl<TP: OperatorTaskProcessor> Operator<TP> {
     /// # Returns
     ///
     /// * `Result<Self, OperatorError>` - The operator.
+    #[allow(clippy::too_many_arguments)]
     pub async fn new(
-        avs_registry_reader: &AvsRegistryChainReader,
         key_pair: &BlsKeyPair,
         operator_address: Address,
         operator_name: &str,
-        client_aggregator: &ClientAggregator,
+        logger: SharedLogger,
         ws_rpc_url: &str,
+        http_rpc_url: &str,
+        registry_coordiator_address: Address,
+        operator_state_retriever_address: Address,
+        aggregator_ip_port_address: String,
```

Comment:
Let's avoid using address since it collides with chain addresses.

```suggestion
        aggregator_ip_port: String,
```